src/Classes/AbstractOrFinalRule.php

```diff
@@ -33,6 +48,7 @@ public function getNodeType(): string
     public function processNode(Node $node, Scope $scope): array
     {
         if (!isset($node->namespacedName)
```

Comment:
I usually prefer a safer check:
```php
        if (!$node instance of  Node\Stmt\Class_
            || \in_array($node->namespacedName->toString(), $this->excludedClassNames, true)
            || $node->isAbstract()
            || $node->isFinal()
        ) {
```

Reply:
Sounds good, added - and throwing a `PHPStand\ShouldNotHappenException`.